mapper/javabean/src/main/java/org/hibernate/search/mapper/javabean/session/SearchSession.java

```diff
@@ -34,7 +68,7 @@
 	 * @return The initial step of a DSL where the search query can be defined.
 	 * @see SearchQuerySelectStep
 	 */
-	default <T> SearchQuerySelectStep<?, EntityReference, ?, ?, ?, ?> search(Class<T> type) {
+	default <T> SearchQuerySelectStep<?, EntityReference, T, ?, ?, ?> search(Class<T> type) {
```

Comment:
A good change does not affect any functionality, but it is very important. I don't want to cast type when search is used. It's hard to do the rest of this task without it.

Reply:
It could be a bug that makes the module completely unusable, that's not the point. It's not related to what you're trying to fix, so it should not be in the same commit as the rest. At the very least, please isolate the change in a separate commit. Note that commit can happen before the other commits, that's not a problem.